components/txn_types/src/write.rs

```diff
@@ -100,9 +112,26 @@ impl Write {
             write_type: WriteType::Rollback,
             start_ts,
             short_value,
+            has_overlay_rollback: false,
         }
     }
 
+    #[inline]
+    pub fn set_overlay_rollback(mut self, has_overlay_rollback: bool) -> Self {
+        self.has_overlay_rollback = has_overlay_rollback;
+        self
+    }
+
+    #[inline]
+    pub fn is_protected(&self) -> bool {
```

Comment:
I can't find any place using it. And you can use `write.as_ref().is_protected()`.

Reply:
I'm afraid it will introduce several unnecessary value copying. This function might become dead after some change. I'll check it then.